examples/tests/kv_interactive/main.c

```diff
@@ -0,0 +1,148 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <console.h>
+#include <kv_system.h>
+
+#define KEY_LEN  64
+#define DATA_LEN 64
+
+uint8_t key_buf[KEY_LEN];
+uint8_t data_buf[DATA_LEN];
+uint8_t value_buf[DATA_LEN];
+
+char read_buf[DATA_LEN];
+
+static int get_command(void) {
+  int idx = 0;
+  memset(read_buf, 0, 64);
+
+  // Read in characters until a new line
+  while (1) {
+    int c = getch();
+
+    if (c == RETURNCODE_FAIL) {
+      printf("\ngetch() failed!\n");
+      return 0;
+
+    } else {
+      char in = c;
+
+      if (in == '\n' || in == '\r') {
+        return 0;
+
+      } else {
+        // If this is a valid number record it
+        read_buf[idx] = in;
+        idx += 1;
+
+        // If our buffer is full, quit
+        if (idx >= 64) {
+          return 0;
+        }
+      }
+    }
+  }
+}
+
+static int find_end(int start) {
+  for (int i = start; i < DATA_LEN; i++) {
+    if (read_buf[i] == ' ' || read_buf[i] == '\n' || read_buf[i] == '\0') {
+      return i;
+    }
+  }
+  return DATA_LEN;
+}
+
+int main(void) {
+  int ret;
+
+  printf("[KV Interactive] Text interface to KV store\n");
+  printf("\n");
+  printf("Please enter a command: get, set, or delete\n");
+  printf("   get <key>\n");
+  printf("   set <key> <value>\n");
+  printf("   delete <key>\n");
+  printf("\n");
+
+  while (1) {
+    ret = get_command();
+    if (ret < 0) {
```

Comment:
How can this be less then 0?

Reply:
The internal `getch()` can fail; though get_command had been suppressing that.. https://github.com/tock/libtock-c/pull/325/commits/c03e109400a25f3d1b526c861ee27dd33df58e89, fixed.